addon/batcher.ts

```diff
@@ -1,19 +1,57 @@
-import { buildWaiter, ITestWaiter, Token } from 'ember-test-waiters';
+import { ITestWaiter, Token, buildWaiter } from 'ember-test-waiters';
+
+import { DEBUG } from '@glimmer/env';
 
 type MaybeRequestAnimationFrame = (callback: FrameRequestCallback | Function) => number;
 type DomOperation = [Token, () => void];
 
+const IS_BROWSER = typeof window === 'object' && typeof document === 'object';
 const SCHEDULE_MACROTASK = (callback: Function) => setTimeout(callback);
-const scheduleFn: MaybeRequestAnimationFrame =
-  typeof window === 'object' && typeof window.requestAnimationFrame === 'function'
-    ? window.requestAnimationFrame
-    : SCHEDULE_MACROTASK;
 const readDOMWaiter: ITestWaiter = buildWaiter('ember-batcher: readDOM');
 const mutateDOMWaiter: ITestWaiter = buildWaiter('ember-batcher: mutateDOM');
 
 const reads: Array<DomOperation> = [];
 const mutations: Array<DomOperation> = [];
+let visibilityChange: Function = () => {};
 let running: boolean = false;
+let scheduleFnExecuted: boolean = false;
+
+const rafRace = (callback: Function) => {
+  setTimeout(() => {
+    if (!scheduleFnExecuted) {
+      callback();
+    }
+  }, 20);
+
+  return requestAnimationFrame(() => {
+    scheduleFnExecuted = true;
+    callback();
```

Comment:
If the timer fires, we should either cancel the raf or guard against it invoking callback

Reply:
Ya good call. I'll add.

Reply:
Done. I've mimicked the guarded `scheduleFn` execution for both.